### PR TITLE
[FIX] web: fix missing border-radius on arrow button

### PR DESCRIPTION
**Pull request by chgo-odoo**
This PR fixes an issue about the last arrow button not taking the
`o_first` class  in some cases.

With Odoo 17, we introduced a new behavior for our arrow buttons.
This behavior includes an invisible element placed before and after the
first arrow button on the left and right side.

While this is working fine as long as you don't have any hidden item
insie a dropdown menu, once you trigger a stage that was folded into a
dropdown menu, this invisible element appears on the right side.

Since we use the `o_first` class that sets a `border-radius` on the first
element on the right side, this invisible element was taking that
`o_first` class.

To fix the issue, we search for the first element within the scope that
does not have a `d-none` class, and we then add the class to that specific
element.

task-3628288

### CODE REVIEW

**Comment by chgo-odoo** [comment]
Hello @mano-odoo,

Could you have a look at that small PR, please?

I have set you up the `Office Design` project for testing purpose, but feel free to refer to the task to test yourself or grasp everything from the issue.

Cheers !

**Comment by chgo-odoo** [comment]
Hello @odoo/rd-framework-js team !

Could you have a look at this small PR? Just want to ensure the change is fine.

Thanks !

**Comment by SimonGenin** [comment]
Hello @chgo-odoo, it seems to us a bit weird that the class o-first is added in a function called `areItemsWrapping`. By a quick look, `adjustVisibleItems` feels more natural. 
I can't say for sure cause I've never been able to reproduce the missing border radius though.. So can you fix it in a more suitable way, or provide me with an exact steps to reproduce ? Thank you !

**Comment by chgo-odoo** [comment]
Hi @SimonGenin, apologize, it's true that you might miss some context here. About the location of the change, I put it here since it already had the element I needed to target but I'm all hear if it's better to put it elsewhere, I don't have all the optimization/tech background.

TL;DR, here is a quick recording of the issue : 

https://github.com/odoo/odoo/assets/128030743/9e523b6b-2a2b-4ec5-9010-1a7a09e43d30


To reproduce the issue, you simply need to set a stage in a kanban view as folded, create a new stage behind this one, and then open one of the item and change its stage to its current stage to whatever your new stage is named. If you then click on the arrow button to put it in the folded stage, the extra stage you created will be displayed at the end without a border-radius.

**Comment by SimonGenin** [comment]
@chgo-odoo Thank you for the video ! 
I played around a bit, I believe there is a real deeper bug with folded stages. Your fix kinda work with one stage folded, but when there is several, the order they start showing when you switch from one to another is wrong.
So I guess I'll try to fix that and I'll come back to you after

**Comment by SimonGenin** [comment]
@chgo-odoo Hello again. I misspoke when I mentioned further bugs. I just wasn't properly understanding how the status bar was handling the folded stages.

I made the fix differently and added a test to make sure.

@FrancoisGe or @aab-odoo if it's ok for you we can merge.

**Review by aab-odoo — COMMENTED** [review]
@SimonGenin The commit message hasn't been updated right?

**Comment by SimonGenin** [comment]
@aab-odoo it has not, it's still valid

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by chgo-odoo** [comment]
Hi @SimonGenin, joining the party a bit late but thanks for your investigation and your fix !
